File: Sources/SkipLib/Skip/Bool.kt
// Copyright 2023–2026 Skip
// SPDX-License-Identifier: MPL-2.0
package skip.lib

fun Boolean.random(using: InOut<RandomNumberGenerator>? = null): Boolean {
    return (using?.value ?: systemRandom).next() % 2UL == 0UL
}

File: Sources/SkipLib/Bool.swift
// Copyright 2023–2026 Skip
// SPDX-License-Identifier: MPL-2.0
// SKIP SYMBOLFILE

#if SKIP

extension Bool: CustomStringConvertible {
    public static func random() -> Bool { fatalError() }
    public static func random(using gen: inout RandomNumberGenerator) -> Bool { fatalError() }
    public var description: String { return "" }

    @available(*, unavailable)
    public func toggle() {
    }
}

#endif
